gon/base.py

```diff
@@ -3,31 +3,30 @@
 
 from reprit.base import generate_repr
 
-from .hints import Scalar
 from .utils import validate_value
 
 
 class Point:
     __slots__ = ('_x', '_y')
 
-    def __new__(cls, x: Scalar, y: Scalar) -> 'Point':
+    def __new__(cls, x: Real, y: Real) -> 'Point':
         validate_value(x)
         validate_value(y)
         return super().__new__(cls)
 
-    def __init__(self, x: Scalar, y: Scalar) -> None:
+    def __init__(self, x: Real, y: Real) -> None:
         self._x = x
         self._y = y
 
     @property
-    def x(self) -> Scalar:
+    def x(self) -> Real:
```

Comment:
we can extract this to `hints` called `Coordinate` like in `hypothesis_geometry`:
https://github.com/lycantropos/hypothesis_geometry/blob/master/hypothesis_geometry/hints.py#L8
and then reuse it everywhere